courses/api/views.py

```diff
@@ -251,11 +264,79 @@ class RandomDataViewSet(ReadOnlyCoursesAbstractViewSet):
     }
     """
     serializer_class = RandomDataSerializer
-    filter_fields = ['title', 'description', 'slug', 'data_type', 'paid']
+    filter_fields = ['data_type', 'paid']
+    search_fields = ['title','description','slug']
     ordering_fields = ['data_type', 'title', 'paid']
     ordering = ['data_type']
     queryset = RandomData.objects.all()
 
     def get_queryset(self):
         queryset = RandomData.objects.filter(is_active=True)
         return queryset
+
+
+
+class LimitPagination(MultipleModelLimitOffsetPagination):
+    default_limit = 15
+
+
+class GlobalSearchAPIViewSet(ObjectMultipleModelAPIViewSet):
+    pagination_class = LimitPagination
+
+
+    def get_querylist(self,*args, **kwargs):
+        query = self.request.GET.get('query', None)
+
+        knowledgebase_queryset = KnowledgeBase.objects.filter(Q(title__icontains=query) | 
+                                                      Q(slug__icontains=query) | 
```

Comment:
select_related or prefetch related for lesser queries @Par010 read up on it. Don't implement right now